src/services/ghost/classic-auto-complete/GhostInlineCompletionProvider.ts

```diff
@@ -279,7 +294,10 @@ export class GhostInlineCompletionProvider implements vscode.InlineCompletionIte
 
 			// Once we have enough samples, update the debounce delay to the average
 			const sum = this.latencyHistory.reduce((acc, val) => acc + val, 0)
-			this.debounceDelayMs = Math.round(sum / this.latencyHistory.length)
+			const averageLatency = Math.round(sum / this.latencyHistory.length)
+
+			// Clamp the debounce delay between MIN and MAX
+			this.debounceDelayMs = Math.max(MIN_DEBOUNCE_DELAY_MS, Math.min(averageLatency, MAX_DEBOUNCE_DELAY_MS))
```

Comment:
Sanity: Didi you consider some function of  averageLatency other than averageLatency?  e.g. why not averageLatency\*0.8+100 or averageLatency\*1.2 or whatever?  Probably worth at least a few minutes of thought as to what works best and why. (identity fine, too).

Reply:
Also, worth manual testing with faked slowdowns.